test/SourceKit/CodeExpand/code-expand-multiple-trailing-closures.swift

```diff
@@ -0,0 +1,118 @@
+// RUN: %sourcekitd-test -req=expand-placeholder %s | %FileCheck %s
+
+withMulti1Labeled(a: <#T##() -> ()#>, b: <#T##() -> ()#>)
+// CHECK:      withMulti1Labeled {
+// CHECK-NEXT: a: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: b: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: }
+
+withMulti2Unlabled(<#T##() -> ()#>, _: <#T##() -> ()#>)
+// CHECK:      withMulti2Unlabled {
+// CHECK-NEXT: _: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: _: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: }
+
+withMulti3SecondArg(a: <#T##__skip__##() -> ()#>, b: <#T##() -> ()#>)
+// CHECK:      withMulti3SecondArg {
+// CHECK-NEXT: a: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: b: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: }
+
+withMulti4MiddleArg(a: <#T##() -> ()#>, b: <#T##__skip__##() -> ()#>, c: <#T##() -> ()#>)
+// CHECK:      withMulti4MiddleArg {
+// CHECK-NEXT: a: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: b: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: c: { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: }
+
+singleAlreadyExpand(a: { print("hi") }, b: <#T##() -> ()#>)
+// CHECK:      singleAlreadyExpand(a: { print("hi") }) { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+
+nonTrailing1(a: <#T##() -> ()#>, b: { print("hi") })
+// CHECK:      nonTrailing1(a: {
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }, b: { print("hi") })
+
+nonTrailingAndTrailing1(a: <#T##() -> ()#>, b: { print("hi") }, c: <#T##() -> ()#>)
+// CHECK:      nonTrailingAndTrailing1(a: {
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }, b: { print("hi") }) {
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+
+singleNonClosure(a: <#T##Int#>, b: <#T##() -> ()#>)
+// CHECK:      singleNonClosure(a: Int) { 
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+
+nonTrailing2(a: <#T##() -> ()#>, b: <#T##Int#>)
+// CHECK:      nonTrailing2(a: {
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }, b: Int)
+
+nonTrailingAndTrailing2(a: <#T##() -> ()#>, b: <#T##Int#> c: <#T##() -> ()#>)
+// CHECK:      nonTrailingAndTrailing2(a: {
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }, b: Int) {
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+
+withTypes1(a: <#T##(Bool, Int) -> ()#>, b: <#T##() -> Int#>)
+// CHECK:      withTypes1 {
+// CHECK-NEXT: a: { (<#Bool#>, <#Int#>) in
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: b: { () -> Int in
+// CHECK-NEXT: <#code#>
+// CHECK-NEXT: }
+// CHECK-NEXT: }
+
+withTypesAndLabels1(a: <#T##(_ booly: Bool, inty: Int) -> ()#>, b: <#T##(solo: Xyz) -> ()#>)
+// CHECK:      withTypesAndLabels1 {
+// CHECK-NEXT: a: { (booly, inty) in
```

Comment:
This is unrelated to this PR since it's existing behavior, but I feel like most Swift code I've seen doesn't put the parens around the closure params unless they have explicit types. Is there a reason we print them at the moment?

Reply:
I don't think there's a good reason, it's just what falls out of the implementation since if you have types or return type you need them.  I agree we should remove them when not needed.